feat: Shopping list UI overhaul - add label headings per category

Wetzel402 opened the pull request:
<!--
  This template provides some ideas of things to include in your PR description.

  To start, try providing a short summary of your changes in the Title above. We follow Conventional Commits syntax, please ensure your title is prefixed with one of:
  - `feat: `
  - `fix: `
  - `docs: `
  - `chore: `

  If a section of the PR template does not apply to this PR, then delete that section.

  PLEASE READ:
  -------------------------
  Mealie is moving to a regular, automatic release schedule. This means that all PRs should be in a
  stable state, ready for release. This includes:

  - Ensuring new tests have been added to cover new features, or to prevent regressions.
  - Work is fully complete and usable

 -->

## What type of PR is this?
<!--
  Delete any of the following that do not apply:
 -->
- feature

## What this PR does / why we need it:

- Each label now has it's own add button
![image](https://github.com/user-attachments/assets/1670cd9a-f484-4fd7-a687-6f4052603c68)

<!--
  What goal is this change working towards?
  Provide a bullet pointed summary of how each file was changed.
  Briefly explain any decisions you made with respect to the changes.
  Include anything here that you didn't include in *Release Notes*
  above, such as changes to CI or changes to internal methods.

  If there is a UI component to the change, please include before/after images.
-->

## Which issue(s) this PR fixes:

https://github.com/mealie-recipes/mealie/pull/4077

<!--
If this PR fixes one of more issues, list them here.
One per line, like so:
Fixes #123
Fixes #39
-->

## Testing
<!--
  Describe how you tested this change.
-->
Tested in dev container

Wetzel402 committed:
modified:   frontend/pages/group/data/labels.vue
	modified:   frontend/pages/shopping-lists/_id.vue

Wetzel402 committed:
modified:   mealie/schema/labels/multi_purpose_label.py

Kuchenpirat commented on frontend/pages/shopping-lists/_id.vue:
```diff
@@ -56,14 +56,12 @@
-                {{ $globals.icons.tags }}
-              </v-icon>
-            </span>
-            {{ key }}
+            <div class="text-left">
+              <BaseButton create :color="getLabelColor(value[0])" @click="createEditorOpen = true">{{ key }}</BaseButton>
+            </div>
+            <v-divider/>
           </div>
           <draggable :value="value" handle=".handle" delay="250" :delay-on-touch-only="true" @start="loadingCounter += 1" @end="loadingCounter -= 1" @input="updateIndexUncheckedByLabel(key, $event)">
```
Don't know why this `toggleShowChecked` is here. This will propably only get anoying in the future when the button gets a function and people will actually click it. 
```suggestion
          <div class="text-left">
            <BaseButton create :color="getLabelColor(value[0])" @click="createEditorOpen = true">{{ key }}</BaseButton>
          </div>
          <v-divider/>
          <draggable :value="value" handle=".handle" delay="250" :delay-on-touch-only="true" @start="loadingCounter += 1" @end="loadingCounter -= 1" @input="updateIndexUncheckedByLabel(key, $event)">
```

Kuchenpirat commented on frontend/pages/shopping-lists/_id.vue:
```diff
@@ -56,14 +56,12 @@
-            </span>
-            {{ key }}
+            <div class="text-left">
+              <BaseButton create :color="getLabelColor(value[0])" @click="createEditorOpen = true">{{ key }}</BaseButton>
```
Lets have this be the color of the default label: 

```suggestion
            <BaseButton create :color="getLabelColor(value[0]) ? getLabelColor(value[0]) : '#959595'" @click="createEditorOpen = true">{{ key }}</BaseButton>
```

Kuchenpirat reviewed (commented):
Just a few little things on this one. 

Thanks for your work so far! When everything finally rolls into mealie this will certainly improve the shopping list 😊

Kuchenpirat commented:
Let's use "#959595" as default label color. This would be a compromise and should be readable on both light and dark mode

Wetzel402 committed:
modified:   frontend/pages/group/data/labels.vue
	modified:   frontend/pages/shopping-lists/_id.vue
	modified:   mealie/schema/labels/multi_purpose_label.py

Wetzel402 commented:
I just realized when VS Code did an autoformat it caused a lot of changes in github.

Wetzel402 committed:
modified:   frontend/pages/shopping-lists/_id.vue

Kuchenpirat committed:
Merge branch 'mealie-next' into labels

Kuchenpirat reviewed (approved):
Looks all good 👍
